attacks-and-mitigations.md

```diff
@@ -721,6 +741,15 @@ AS therefore MUST provide a way to detect their support for PKCE. Using AS
 metadata according to [@!RFC8414] is RECOMMENDED, but AS MAY instead provide a
 deployment-specific way to ensure or determine PKCE support.
 
+PKCE provides robust protection against CSRF attacks even in presence of an (A3)
+attacker (which can read the authorization response). When `state` is used or an
+ID Token is returned in the authorization response (e.g.,
+`response_type=code+id_token`), the attacker either learns the `state` value and
+can replay it into the forged authorization response, or can extract the `nonce`
+from the ID Token and use it in a new request to the authorization server to
+mint an ID Token with the same `nonce`. The new ID Token can then be used for
```

Comment:
This is interesting. What is the protection against this attack?

Reply:
As far as I see, only PKCE protects against this.

I was about to say that the attacker model (A3) is relatively strong for a CSRF attack; but when we agree that (A3) is not uncommon (which we did for the code misuse discussion), then that would mean we need PKCE for everyone, always.